### PR TITLE
Frontend: changed build tooling from cra to vite

### PR DESCRIPTION
**Pull request by suryavirkapur**
This pull request migrates the project from Create React App to Vite as the build tool. Vite has become the more widely adopted standard for frontend tooling, offering faster development server spin-up, quicker file updates and reflections, improved build performance, and enhanced developer experience. This change aims to leverage these benefits for a more efficient development workflow. Furthermore, I have added a placeholder logo for the project, and updated the icons. 

Additionally, the license year in the project's LICENSE file has been updated to 2024.

### CODE REVIEW

**Review by neubig — CHANGES_REQUESTED** [review]
Hello, thank you for the contribution!

I do think that transitioning to vite would be a good idea, but:
1. Could you please not change the directory name for the frontend, currently the PR is changing it from `frontend` to `client`
2. Could you please add more information to the README on how to get started with frontend development?

**Comment by suryavirkapur** [comment]
Hey, I have addressed the requested changes in the PR:

1. I have reverted the directory name change and kept it as frontend.
2. I have added more information to the README on how to get started with frontend development, including prerequisites, setup instructions, development server details, building for production, running tests, code structure, and additional resources.

The PR is now ready for your review. Please let me know if there's anything else you need from my side.

File: frontend/README.md [hunk 1]
```diff
@@ -1,44 +1,34 @@
...
+1. Clone the repository: `git clone https://github.com/OpenDevin/OpenDevin.git`
+2. Navigate to the frontend directory: `cd OpenDevin/frontend`
+3. Install dependencies: `npm install`
 
-The page will reload if you make edits.\
-You will also see any lint errors in the console.
+### Development Server
 
-### `npm test`
+Run `npm start` to start the development server. Open `http://localhost:5173` to view the application in the browser.
```
Comment (neubig) [inline frontend/README.md:16]: I tried to run this in my local environment and get the following error.

```bash
$ npm start
npm ERR! Missing script: "start"
npm ERR! 
npm ERR! Did you mean one of these?
npm ERR!     npm star # Mark your favorite packages
npm ERR!     npm stars # View packages marked as favorites
npm ERR! 
npm ERR! To see a list of scripts, run:
npm ERR!   npm run
```

**Review by neubig — CHANGES_REQUESTED** [review]
Thank you for the modifications! I downloaded and tried to run the code, and have two requests.

1. Could you please try to limit the changes made to only changes related to the theme of the PR (changing the build tooling to vite). There are also many other changes such as moving the `components` directory to `Components`, adding new icons, adding eslintrc, etc. It's too much work for me to review all the changes at once.
2. I had trouble running the code, please check my comment below.

**Comment by suryavirkapur** [comment]
Right! I apologize for the confusion caused by the additional changes. I will address your requests one by one:

1. I understand your concern about the scope of the changes. I will focus solely on updating the build tooling to Vite and avoid making any unrelated modifications to the project structure and components. 
2. I also made unrelated modifications to the project configuration files. I have reverted these too. So you should no longer have problems running the code. Kindly go through the README again.

**Comment by neubig** [comment]
Hi @suryavirkapur , thanks so much again for the patience!

We just [merged in some linting/formatting stuff](https://github.com/OpenDevin/OpenDevin/pull/32) that will standardize our formatting and make this PR easier to review. Could you please `git pull origin main` to merge in these changes, fix any merge conflicts, and I'll review after that?

**Comment by suryavirkapur** [comment]
Thanks for the update! I've pulled the latest changes from origin main, resolved all merge conflicts, and pushed the updated branch. The PR should be ready for review now. Please let me know if you need anything else!

**Comment by neubig** [comment]
No worries, and thanks for bearing with this! We're just setting things up so it's a slight bit chaotic.

**Pull request closed without merging**